packages/api/app/Utils/algolia.js

```diff
@@ -0,0 +1,113 @@
+const Config = use('Adonis/Src/Config');
+const algoliasearch = use('App/Services/AlgoliaSearch');
+
+const algoliaConfig = Config.get('algolia');
+const indexObject = algoliasearch.initIndex(algoliaConfig.indexName);
+
+const { roles } = require('./roles_capabilities');
+
+const CATEGORY_TAXONOMY_SLUG = 'CATEGORY';
+const CLASSIFICATION_TAXONOMY_SLUG = 'CLASSIFICATION';
+const DIMENSION_TAXONOMY_SLUG = 'DIMENSION';
+const TARGET_AUDIENCE_TAXONOMY_SLUG = 'TARGET_AUDIENCE';
+
+/**
+ * Returns all technology terms related to algolia index
+ *
+ * @param {object} technology The technology object
+ * @returns {object} The technology terms to be indexed by algolia
+ */
+const normalizeAlgoliaTechnologyTerms = (technology) => {
+	const defaultTermMasc = 'Não definido';
+	const defaultTermFem = 'Não definida';
+
+	const termsObj = technology.terms.reduce((acc, obj) => {
+		acc[obj.taxonomy.taxonomy] = obj.term;
+		return acc;
+	}, {});
+
+	return {
+		category: termsObj[CATEGORY_TAXONOMY_SLUG] || defaultTermFem,
+		classification: termsObj[CLASSIFICATION_TAXONOMY_SLUG] || defaultTermFem,
+		dimension: termsObj[DIMENSION_TAXONOMY_SLUG] || defaultTermFem,
+		targetAudience: termsObj[TARGET_AUDIENCE_TAXONOMY_SLUG] || defaultTermMasc,
+	};
+};
+
+/**
+ * Returns grouped sum of technology implementation cost and maintenance cost
+ *
+ * @param {object} technology The technology object
+ * @returns {object} The technology costs
+ */
+const normalizeAlgoliaTechnologyCosts = (technology) => {
+	const { costs } = technology.technologyCosts[0];
+
+	const implementationCost = costs
+		.filter((cost) => cost.cost_type === 'implementation_costs')
+		.reduce((acc, curr) => acc + curr.value * curr.quantity, 0);
```

Comment:
`reduce` is mostly used to replace `filter` + `map`. In this case here, we can avoid looping through the array twice, by filtering the data inside `reduce`. Something like
```
const implementationCost = costs
		.reduce((acc, curr) => curr.cost_type === 'implementation_costs' ? acc + curr.value * curr.quantity : acc, 0);

Reply:
Ty @lcnogueira, you're totally right